rubix-prestosql/src/main/java/com/qubole/rubix/prestosql/PrestoClusterManager.java

```diff
@@ -0,0 +1,289 @@
+import com.google.gson.Gson;
+import com.qubole.rubix.common.utils.ClusterUtil;
+import com.qubole.rubix.spi.ClusterManager;
+import com.qubole.rubix.spi.ClusterType;
+import com.qubole.rubix.spi.thrift.ClusterNode;
+import com.qubole.rubix.spi.thrift.NodeState;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+import java.net.InetAddress;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by stagra on 14/1/16.
+ */
+public class PrestoClusterManager extends ClusterManager
+{
+  private int serverPort = 8081;
+  private String serverAddress = "localhost";
+  static LoadingCache<String, List<ClusterNode>> nodesCache;
+
+  private Log log = LogFactory.getLog(PrestoClusterManager.class);
+
+  public static String serverPortConf = "caching.fs.presto-server-port";
+  public static String serverAddressConf = "master.hostname";
+  public static String yarnServerAddressConf = "yarn.resourcemanager.address";
+
+  // Safe to use single instance of HttpClient since Supplier.get() provides synchronization
+  @Override
+  public void initialize(Configuration conf)
+  {
+    super.initialize(conf);
+    this.serverPort = conf.getInt(serverPortConf, serverPort);
+    this.serverAddress = ClusterUtil.getMasterHostname(conf);
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    nodesCache = CacheBuilder.newBuilder()
+        .refreshAfterWrite(getNodeRefreshTime(), TimeUnit.SECONDS)
+        .build(CacheLoader.asyncReloading(new CacheLoader<String, List<ClusterNode>>()
+        {
+          @Override
+          public List<ClusterNode> load(String s)
+              throws Exception
+          {
+            try {
+              URL allNodesRequest = getNodeUrl();
+              URL failedNodesRequest = getFailedNodeUrl();
+
+              HttpURLConnection allHttpCon = (HttpURLConnection) allNodesRequest.openConnection();
+              allHttpCon.setConnectTimeout(500); //ms
+              allHttpCon.setRequestMethod("GET");
+
+              int allNodesResponseCode = allHttpCon.getResponseCode();
+
+              StringBuffer allResponse = new StringBuffer();
+              StringBuffer failedResponse = new StringBuffer();
+              try {
+                if (allNodesResponseCode == HttpURLConnection.HTTP_OK) {
+                  BufferedReader in = new BufferedReader(new InputStreamReader(allHttpCon.getInputStream()));
+                  String inputLine = "";
+                  try {
```

Comment:
You can use **try-with-resources** instead of try-catch-finally for closeable inputstream **in**

Reply:
all old code hence the old standards. We need to modernize the whole project.